app/src/main/java/org/mozilla/focus/tabs/TabsSession.java

```diff
@@ -85,15 +85,16 @@ public int getTabsCount() {
      *
      * @param models
      */
-    public void restoreTabs(List<TabModel> models) {
+    public void restoreTabs(List<TabModel> models, String currentTabId) {
         for (final TabModel model : models) {
             final Tab tab = new Tab(model);
             bindCallback(tab);
             tabs.add(tab);
         }
 
         if (tabs.size() > 0 && tabs.size() == models.size()) {
-            currentIdx = 0; // first tab
+            int currentTabIndex = getTabIndex(currentTabId);
+            currentIdx = currentTabIndex != -1 ? currentTabIndex : 0;
```

Comment:
`(currentTabIndex != -1)` would be better. Since it is just a local variable in a small scope, we can make it shorter to be `final int index = getTabIndex(currentTabId);`